packages/engine/paima-runtime/src/cde-config/utils.ts

```diff
@@ -1,15 +1,37 @@
 import * as fs from 'fs/promises';
 
-import { doLog } from '@paima/utils';
+import { ChainDataExtensionType, doLog } from '@paima/utils';
 
 import type { ChainDataExtension } from '@paima/sm';
 
 export function getEarliestStartBlockheight(config: ChainDataExtension[]): number {
-  const startBlockheights = config.map(cde => cde.startBlockHeight).filter(sbh => !!sbh);
+  const startBlockheights = config
+    .map(cde => {
+      if (cde.cdeType != ChainDataExtensionType.CardanoPool) {
+        return cde.startBlockHeight;
+      } else {
+        return null;
+      }
+    })
+    .filter(sbh => !!sbh) as number[];
   const minStartBlockheight = Math.min(...startBlockheights);
   return isFinite(minStartBlockheight) ? minStartBlockheight : -1;
```

Comment:
All of this should probably be a single `.reduce` call instead (same with the function below)